File: python/modules/class_implemented_derived.py
from i_interface_name import IInterface
from class_abstract_base import ClassAbstractBase

class ClassImplementedDerived(ClassAbstractBase):
    def __init__(self):

        super().__init__()

        self._my_name = "ClassImplementedDerived"

        self._related_child_member = ""

    def __del__(self):
        # Destructor: cleanup resources if needed.
        pass

    def start(self) -> IInterface.STATUS:
        # Override start method.
        # Add your implementation here.
        return IInterface.STATUS.OK_STARTED

    def stop(self) -> IInterface.STATUS:
        # Override stop method.
        # Add your implementation here.
        return IInterface.STATUS.OK_STOPPED

    def _related_child_method(self) -> IInterface.STATUS:
        # Private helper method.
        # Add your implementation here.
        pass

File: cpp/src/class_implemented_derived.cpp
#include "class_implemented_derived.h"

#include <iostream>

ClassImplementedDerived::ClassImplementedDerived() {
    this->_my_name = "ClassImplementedDerived";
    this->_related_child_member = "related child member";
}

ClassImplementedDerived::~ClassImplementedDerived() {}

ClassImplementedDerived::STATUS ClassImplementedDerived::start() {

    ClassImplementedDerived::STATUS status = ClassImplementedDerived::STATUS::OK_STARTED;

    bool isStarted = true;

    if (!isStarted) {
        status = ClassImplementedDerived::STATUS::ERROR_CANNOT_START;
    }

    std::cout << this->_my_name << " :: " << this->getStatusMessage(status) << std::endl;

    return status;
}

ClassImplementedDerived::STATUS ClassImplementedDerived::stop() {
    
    ClassImplementedDerived::STATUS status = ClassImplementedDerived::STATUS::OK_STOPPED;

    bool isStopped = true;

    if (!isStopped) {
        status = ClassImplementedDerived::STATUS::ERROR_CANNOT_STOP;
    }

    std::cout << this->_my_name << " :: " << this->getStatusMessage(status) << std::endl;

    return status;
}